### PR TITLE
COMMERCE-7742 Shipment Options appearing for Virtual Products

### PR DESCRIPTION
**Pull request by christian-chiappa**

### CODE REVIEW

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp [hunk 1]
```diff
@@ -160,23 +160,31 @@ boolean shippable = BeanParamUtil.getBoolean(cpDefinition, request, "shippable",
...
 
-				<aui:input name="height" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_DIMENSION)) %>" />
+					<aui:input name="height" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_DIMENSION)) %>" />
 
-				<aui:input name="depth" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_DIMENSION)) %>" />
+					<aui:input name="depth" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_DIMENSION)) %>" />
 
-				<aui:input name="weight" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_WEIGHT)) %>" />
+					<aui:input name="weight" suffix="<%= HtmlUtil.escape(cpDefinitionConfigurationDisplayContext.getCPMeasurementUnitName(CPMeasurementUnitConstants.TYPE_WEIGHT)) %>" />
+				</div>
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp:187]: Why these need to be shippable otpion?

File: modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		frutillaRule.scenario(
+			"Add product definition"
+		).given(
+			"I add a product definition"
+		).when(
+			"ignoreSKUCombinations is true"
+		).and(
+			"hasDefaultInstance is true"
+		).and(
+			"shippable is true"
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java:98]: Why you don't add a clause for virtual product in this scenario?

File: modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+		CPDefinition cpDefinition = CPTestUtil.addCPDefinition(
+			_commerceCatalog.getGroupId(), VirtualCPTypeConstants.NAME);
+
+		long cpDefinitionId = cpDefinition.getCPDefinitionId();
+
+		Date displayDate = cpDefinition.getDisplayDate();
+		Date expirationDate = cpDefinition.getExpirationDate();
+
+		_cpDefinitionLocalService.updateCPDefinition(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java:135]: so you are preventing to make a virtual product to be shippable, right?

**Comment by andrea-ale-sbarra** [comment]
@luca-pellizzon beside the question I made this look good to me.  I'll wait for answer @christian-chiappa before sending this to brian

File: modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java (hunk 3 above)
Comment (christian-chiappa) [reply modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java:135]: Yes, a virtual product cannot be shipped, so it makes no sense to create a virtual product with shippable true

File: modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp (hunk 1 above)
Comment (christian-chiappa) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_configuration.jsp:187]: If a product is not shippable, I don't need to show package's info input fields(height, width...)
If a product is virtual, you cannot toggle the shippable option

File: modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java (hunk 2 above)
Comment (christian-chiappa) [reply modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java:98]: The test is inside a class that is called VirtualTest, so I thought is was not necessary to specify it. I can add it if you want

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 4]
```diff
@@ -1585,10 +1592,17 @@ public CPDefinition updateCPDefinition(
...
 		validate(
 			groupId, ddmStructureKey, metaTitleMap, metaDescriptionMap,
-			metaKeywordsMap, displayDate, expirationDate,
-			cpDefinition.getProductTypeName());
+			metaKeywordsMap, displayDate, expirationDate, productTypeName);
+
+		if (StringUtil.equalsIgnoreCase(
+				productTypeName, VirtualCPTypeConstants.NAME)) {
+
+			shippable = false;
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1604]: same here

File: modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java (hunk 2 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java:98]: Got it, but it's better to add it, thanks!

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Added suggestions for changes

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 5]
```diff
@@ -193,6 +194,12 @@ public CPDefinition addCPDefinition(
 		CProduct cProduct = cProductLocalService.addCProduct(
 			externalReferenceCode, groupId, userId, new ServiceContext());
 
+		if (StringUtil.equalsIgnoreCase(
+				productTypeName, VirtualCPTypeConstants.NAME)) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:198]: Ehy @christian-chiappa I believe it is better not to change value of signature variables but rather do something like this

```suggestion
if (StringUtil.equalsIgnoreCase(productTypeName, VirtualCPTypeConstants.NAME)) {
    cpDefinition.setShippable(false);
} else {
    cpDefinition.setShippable(shippable);
}
```

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
@andrea-ale-sbarra can you check if all the changes you requested have been met so that we can forward this?
Thanks

**Pull request closed without merging**